### PR TITLE
CES Changes 

### PR DESCRIPTION
**Pull request by jayeshh123**
This PR is related to CES changes.

### CODE REVIEW

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf [hunk 1]
```diff
@@ -370,3 +371,97 @@ variable "scale_encryption_admin_password" {
...
+
+variable "total_client_cluster_instances" {
+  type        = number
+  default     = 2
+  description = "Client cluster node counts"
+}
+
+variable "client_vsi_osimage_name" {
+  type        = string
+  default     = "ibm-redhat-8-6-minimal-amd64-6"
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: ibm-redhat-8-6-minimal-amd64-6 -> ibm-redhat-8-8-minimal-amd64-2

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 2]
```diff
@@ -21,14 +21,15 @@ locals {
...
+  scale_ces_enabled            = var.total_protocol_cluster_instances > 0 ? true : false
+  enable_sec_interface_compute = local.scale_ces_enabled == false && data.ibm_is_instance_profile.compute_profile.bandwidth[0].value >= 64000 ? true : false
+  enable_sec_interface_storage = local.scale_ces_enabled == false && var.storage_type != "persistent" && data.ibm_is_instance_profile.storage_profile.bandwidth[0].value >= 64000 ? true : false
   enable_mrot_conf             = local.enable_sec_interface_compute && local.enable_sec_interface_storage ? true : false
 }
 
 module "generate_compute_cluster_keys" {
   source  = "../../../resources/common/generate_keys"
-  turn_on = var.total_compute_cluster_instances > 0 ? true : false
+  turn_on = var.total_client_cluster_instances > 0 || var.total_compute_cluster_instances > 0 ? true : false
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Have separate keys for compute & client clusters.

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 3]
```diff
@@ -199,11 +206,17 @@ data "ibm_is_image" "compute_instance_image" {
...
+  name  = var.client_vsi_osimage_name
+  count = var.total_client_cluster_instances > 0 ? 1 : 0
+}
+
 data "ibm_is_subnet" "compute_cluster_private_subnets_cidr" {
   identifier = var.vpc_compute_cluster_private_subnets[0]
 }
 
 module "compute_cluster_instances" {
+  count                        = var.total_compute_cluster_instances > 0 ? 1 : 0
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Check the necessity of count.

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf [hunk 4]
```diff
@@ -230,6 +243,45 @@ module "compute_cluster_instances" {
   enable_sec_interface_compute = local.enable_sec_interface_compute
 }
 
+module "client_cluster_instances" {
+  count                        = var.total_client_cluster_instances > 0 && local.scale_ces_enabled == true ? 1 : 0
```
Comment (digvijay-ukirde) [inline ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: Check the necessity of count.

File: ibmcloud_scale_templates/sub_modules/instance_template/variables.tf (hunk 1 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/variables.tf]: resolved

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 2 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: resolved

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 3 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: resolved

File: ibmcloud_scale_templates/sub_modules/instance_template/main.tf (hunk 4 above)
Comment (jayeshh123) [reply ibmcloud_scale_templates/sub_modules/instance_template/main.tf]: resolved

**Review by digvijay-ukirde — COMMENTED** [review]
Looks good.
@sasikeda Can you please review and merge the changes.

**Comment by digvijay-ukirde** [comment]
TODO: Update the IBMCloud public endpoints to private in the following PR.

**Pull request merged**